src/Filter/Model/FilterData.php

```diff
@@ -13,9 +13,6 @@
 
 namespace Sonata\AdminBundle\Filter\Model;
 
-/**
- * @psalm-immutable
```

Comment:
Why you removed this?

Reply:
Because currently it's not really immutable
```
52     @readonly property Sonata\AdminBundle\Filter\Model\FilterData::$type   
         is assigned outside of the constructor.                                
  56     @readonly property Sonata\AdminBundle\Filter\Model\FilterData::$value  
         is assigned outside of the constructor.                                
  57     @readonly property                                                     
         Sonata\AdminBundle\Filter\Model\FilterData::$hasValue is assigned      
         outside of the constructor.          
```

Reply:
https://psalm.dev/docs/annotating_code/supported_annotations/#psalm-immutable
and https://psalm.dev/docs/annotating_code/supported_annotations/#psalm-readonly-and-readonly

Reply:
I'm not sure if this is a bug in psalm, those properties are accessed in a named constructor, that doesn't make the class mutable